lib/queuePopulator/LogReader.js

```diff
@@ -355,8 +358,17 @@ class LogReader {
         if (logRes.info.start === null) {
             return done();
         }
-        batchState.nextLogOffset =
-            logRes.info.start + logStats.nbLogRecordsRead;
+
+        // Hopefully there is a better way to find out if dmd, file, or MongoDB
+        if (this.logConsumer.mongoUrl) {
+            // We need to store both the timestamp and uniq Oplog id for MongoDB
+            // oplogs, so we concat the string to be '${timestamp}|${uniqID}'
+            batchState.nextLogOffset =
```

Comment:
in `_readLogOffset` we use `Number.parseInt` to fetch the saved log offset. I'm thinking this can cause issues when reading the log offset in this format.

Reply:
Should we not be using `data.toString()` instead of `Number.parseInt(data, 10)` in `_readLogOffset`?

_NOTE_: we use  `this.logOffset.toString();` in `_writeLogOffset()`?

Also  `Number.parseInt("6544059656791130000|-8572070413137960701", 10)` returns `6544059656791130000` so, for MongoDB, it seems that we are reading the entire logOffset because of `Number.parseInt`. @jonathan-gramain Do you know any reason why we are using `Number.parseInt` and no `.toString();`

Reply:
As long as we don't have to increment/decrement the value, but only forward the value as-is from the source, it should be good to use `toString`. I don't recall the exact reason why we're using `parseInt` originally but there may be none particular.

Note that it should also be possible to use a serialized JSON object instead of a custom separator, and use `JSON.parse()` in place of `Number.parseInt`, it would be more extensible and self-documented, and we should be backward-compatible since a serialized integer is a valid JSON object.